database/src/migrations/20210611115350_block_observation.ts

```diff
@@ -0,0 +1,95 @@
+import Knex from 'knex';
+
+const DB_SCHEMA = process.env.DB_SCHEMA;
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`
+
+  GRANT ALL ON SCHEMA biohub TO postgres;
+
+  set schema '${DB_SCHEMA}';
+  set search_path = ${DB_SCHEMA},public;
+
+--
+-- TABLE: block_observation
+--
+
+CREATE TABLE ${DB_SCHEMA}.block_observation(
+    id                integer           GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1),
+    b_id              integer           NOT NULL,
+    s_id              integer           NOT NULL,
+    start_time        timestamptz(6)    DEFAULT now() NOT NULL,
+    end_time          timestamptz(6)    DEFAULT now() NOT NULL,
+    observation_cnt   integer           NOT NULL,
+    data              json              NOT NULL,
+    create_date       timestamptz(6)    DEFAULT now() NOT NULL,
+    create_user       integer           NOT NULL,
+    update_date       timestamptz(6),
+    update_user       integer,
+    revision_count    integer           DEFAULT 0 NOT NULL,
+    CONSTRAINT "PK170" PRIMARY KEY (id)
+)
+;
+
+
+    COMMENT ON COLUMN block_observation.id IS 'System generated surrogate primary key identifier.'
+    ;
+    COMMENT ON COLUMN block_observation.b_id IS 'System generated surrogate primary key identifier.'
+    ;
+    COMMENT ON COLUMN block_observation.start_time IS 'The datetime the observation was started.'
```

Comment:
is this really a datetime or is it gonna be just a timestamp? same question for the end_time

Reply:
good question ..

Reply:
we're going to use the 'time' data type.  We just need to make sure it can be converted to ISO 8601.